File: 2. Structural Design Patterns/Adapter/Go/AdapterTemplate.go
package main

// Adapter Pattern – Go Template
// Convert Adaptee to Target via Adapter.

type Target interface{ Request() }

type Adaptee struct{}
func (Adaptee) SpecificRequest() { /* ... */ }

type Adapter struct{ a Adaptee }
func (ad Adapter) Request(){ ad.a.SpecificRequest() }

func main(){ var t Target = Adapter{a: Adaptee{}}; t.Request() }

File: 2. Structural Design Patterns/Adapter/C#/AdapterTemplate.cs
// Adapter Pattern – C# Template
// Convert the interface of a class into another interface clients expect.

namespace AdapterTemplate
{
    interface ITarget { void Request(); }

    class Adaptee { public void SpecificRequest() { /* ... */ } }

    class Adapter : ITarget
    {
        private readonly Adaptee _adaptee = new Adaptee();
        public void Request() => _adaptee.SpecificRequest();
    }

    class Program { static void Main(){ ITarget target = new Adapter(); target.Request(); } }
}
